app/src/main/java/com/babylon/wallet/android/presentation/settings/personas/createpersona/CreatePersonaScreen.kt

```diff
@@ -70,38 +64,29 @@ fun CreatePersonaScreen(
     viewModel: CreatePersonaViewModel,
     onBackClick: () -> Unit,
     modifier: Modifier = Modifier,
-    onContinueClick: (
-        personaId: IdentityAddress
-    ) -> Unit = {},
+    onContinueClick: () -> Unit = {},
 ) {
     val state by viewModel.state.collectAsStateWithLifecycle()
-    if (state.loading) {
-        FullscreenCircularProgressContent()
-    } else {
-        CreatePersonaContent(
-            state = state,
-            onPersonaNameChange = viewModel::onDisplayNameChanged,
-            onPersonaCreateClick = viewModel::onPersonaCreateClick,
-            onBackClick = onBackClick,
-            modifier = modifier,
-            onSelectionChanged = viewModel::onSelectionChanged,
-            onAddFields = viewModel::onAddFields,
-            onDeleteField = viewModel::onDeleteField,
-            onValueChanged = viewModel::onFieldValueChanged,
-            onFieldFocusChanged = viewModel::onFieldFocusChanged,
-            onPersonaDisplayNameFocusChanged = viewModel::onPersonaDisplayNameFieldFocusChanged,
-            onAddFieldSheetVisible = viewModel::setAddFieldSheetVisible,
-            onMessageShown = viewModel::onMessageShown
-        )
-    }
+    CreatePersonaContent(
+        state = state,
+        onPersonaNameChange = viewModel::onDisplayNameChanged,
+        onPersonaCreateClick = viewModel::onPersonaCreateClick,
+        onBackClick = onBackClick,
+        modifier = modifier,
+        onSelectionChanged = viewModel::onSelectionChanged,
+        onAddFields = viewModel::onAddFields,
+        onDeleteField = viewModel::onDeleteField,
+        onValueChanged = viewModel::onFieldValueChanged,
+        onFieldFocusChanged = viewModel::onFieldFocusChanged,
+        onPersonaDisplayNameFocusChanged = viewModel::onPersonaDisplayNameFieldFocusChanged,
+        onAddFieldSheetVisible = viewModel::setAddFieldSheetVisible,
+        onMessageShown = viewModel::onMessageShown
+    )
 
-    LaunchedEffect(Unit) {
-        viewModel.oneOffEvent.collect { event ->
-            when (event) {
-                is CreatePersonaEvent.Complete -> onContinueClick(
-                    event.personaId
-                )
-            }
+    LaunchedEffect(state.shouldNavigateToCompletion) {
```

Comment:
I used state here instead of one off event, since when persona is created by corresponding sheet, events that are propagated to `CreatePersonaViewModel` are received when bottom sheet is still on the screen, and navigation to completion screen was not being triggered.